Gems/AWSMetrics/cdk/aws_metrics/aws_utils/resource_name_sanitizer.py

```diff
@@ -0,0 +1,42 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+import hashlib
+
+MAX_RESOURCE_NAME_LENGTH_MAPPING = {
+    'athena_work_group': 128,
+    'athena_named_query': 128,
+    'cloudformation_stack': 128,
+    'cloudwatch_dashboard': 255,
+    'cloudwatch_log_group': 512,
+    'firehose_delivery_stream': 64,
+    'iam_managed_policy': 144,
+    'iam_role': 64,
+    'kinesis_application': 128,
+    'kinesis_stream': 128,
+    'lambda_function': 64,
+    's3_bucket': 63
+}
+
+
+def sanitize_resource_name(resource_name: str, resource_type: str) -> str:

```

Comment:
Do you want to sanitize for invalid characters as well? Names should use only letters, numbers, hyphens, or underscores with no spaces

Reply:
That makes sense to me. I'll add some validations here if invalid characters are detected.

Reply:
Talked offline. I'll update comments for the sanitization function.